### PR TITLE
chore(MeshService): rename `meshServices.enabled` to `meshServices.mode`

### LINKED ISSUE #11585: Change `Mesh.meshServices.enabled` to `Mesh.meshServices.mode` or `type`
### What happened?

```
  message MeshServices {
    Enabled enabled = 1;

    enum Enabled {
      // MeshServices are't generated
      Disabled = 0;
      // MeshServices are generated and used for configuration
      Everywhere = 1;
      // MeshServices are generated but only used for configuration where
      // configured via reachableBackends
      ReachableBackends = 2;
      // MeshServices are generated, used for configuration and kuma.io/services
      // are not used
      Exclusive = 3;
    }
  }
```
`meshService.enabled: Disabled` is very weird.

`meshServices.mode: Disabled|Everywhere|ReachableBackends|Exclusive` is better. Or should it be type?

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues -- Closes #11585 
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/framework/setup.go [hunk 1]
```diff
@@ -175,21 +175,21 @@ spec:
 	return YamlK8s(mesh)
 }
 
-func MTLSMeshWithMeshServicesKubernetes(name string, meshServicesEnabled string) InstallFunc {
```
Comment (lukidzi) [inline test/framework/setup.go:178]: Can we remove them to use `samples.MeshBuilder()...` in the test?

Comment (michaelbeaumont) [reply test/framework/setup.go:178]: It's completely unrelated to this and shouldn't be in this PR

Comment (michaelbeaumont) [reply test/framework/setup.go:178]: we have https://github.com/kumahq/kuma/issues/11294

Comment (michaelbeaumont) [reply test/framework/setup.go:178]: done

**Review by Automaat — APPROVED** [review]

**Pull request merged**